### PR TITLE
8313256: Exclude failing multicast tests on AIX

### PR DESCRIPTION
**Pull request by TOatGithub**
exclude multicast and related tests on AIX mentioned in https://bugs.openjdk.org/browse/JDK-8308807 until this is fixed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8313256](https://bugs.openjdk.org/browse/JDK-8313256): Exclude failing multicast tests on AIX (**Sub-task** - P4)

### Reviewers
 * [Christoph Langer](https://openjdk.org/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15074/head:pull/15074` \
`$ git checkout pull/15074`

Update a local copy of the PR: \
`$ git checkout pull/15074` \
`$ git pull https://git.openjdk.org/jdk.git pull/15074/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15074`

View PR using the GUI difftool: \
`$ git pr show -t 15074`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15074.diff">https://git.openjdk.org/jdk/pull/15074.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15074#issuecomment-1655946051)

### CODE REVIEW

File: test/jdk/ProblemList.txt [hunk 1]
```diff
@@ -789,6 +793,7 @@ java/awt/FullScreen/TranslucentWindow/TranslucentWindow.java 8258103 linux-all
 java/awt/Focus/FrameMinimizeTest/FrameMinimizeTest.java 8016266 linux-x64
 
 ############################################################################
-# java/foreign
+
+# jdk_foreign
```
Comment (tstuefe) [inline test/jdk/ProblemList.txt:797]: Please remove this hunk

File: test/jdk/ProblemList.txt [hunk 2]
```diff
@@ -528,12 +529,15 @@ javax/management/remote/mandatory/subjectDelegation/SubjectDelegation1Test.java
...
-java/net/MulticastSocket/NoLoopbackPackets.java                 7122846 macosx-all
-java/net/MulticastSocket/SetLoopbackMode.java                   7122846 macosx-all
+java/net/MulticastSocket/NoLoopbackPackets.java                 7122846,8308807 macosx-all,aix-ppc64
+java/net/MulticastSocket/SetLoopbackMode.java                   7122846,8308807 macosx-all,aix-ppc64
 
-java/net/MulticastSocket/Test.java                              7145658 macosx-all
+java/net/MulticastSocket/Test.java                              7145658,8308807 macosx-all,aix-ppc64
 
-java/net/ServerSocket/AcceptInheritHandle.java                  8211854 aix-ppc64
+java/net/ServerSocket/AcceptInheritHandle.java                  8211854,8308807 aix-ppc64
```
Comment (tstuefe) [inline test/jdk/ProblemList.txt:537]: Needed? Test was not mentioned in JDK-8308807.

**Review by tstuefe — CHANGES_REQUESTED** [review]
Hi Thomas,

seems okay mostly, small nits remain, see inline remarks.

BTW you can test this on AIX via `jtreg -exclude:<problemlist file> -l <tests>`. [1] I often do this since I keep getting the Problemlist syntax wrong.

Cheers, Thomas

[1] https://openjdk.org/jtreg/command-help.html

File: test/jdk/ProblemList.txt (hunk 1 above)
Comment (AlanBateman) [reply test/jdk/ProblemList.txt:797]: The heading on the sections are usually test group names so "jdk_foreign" is correct, but I don't know what this change is doing in this PR.

Comment (TOatGithub) [reply test/jdk/ProblemList.txt:797]: Thanks guys for pointing this out to a newbie! Would you expect me to add a NOOP PR with correct labeling?
Regards
Thomas

File: test/jdk/ProblemList.txt (hunk 2 above)
Comment (TOatGithub) [reply test/jdk/ProblemList.txt]: Indeed, this was a mistake - added a commit to revert: (0fe3eaecf3f39e2905a6f9b58e6736fc7de583e1)
Thomas

File: test/jdk/ProblemList.txt (hunk 1 above)
Comment (tstuefe) [reply test/jdk/ProblemList.txt:797]: > Thanks guys for pointing this out to a newbie! Would you expect me to add a NOOP PR with correct labeling?
> Regards
> Thomas

You can if you want to. I'm unemotional :-)

If you do, you can mark it as "trivial" (as in, mention in the PR description its a "Trivial" change, which - when accepted by reviewers - means just one review is needed and it is exempt from the usual "wait 24hrs before pushing" rule.

Comment (TOatGithub) [reply test/jdk/ProblemList.txt:797]: So if @AlanBateman does not insist, I'd keep it as is and hopefully rember for next time ;-)

Comment (AlanBateman) [reply test/jdk/ProblemList.txt:797]: I don't have a string opinion either way. I think the main thing is that I suspect you'll need to exclude more tests as there are several tests for DatagramChannel that join multicast groups and some of these likely assume they can join an IPv4 multicast group when cases where the DatagramChannel is created with the no-arg open method (and IPv6 is enabled) or with the IPV6 protocol family.

Comment (RealCLanger) [reply test/jdk/ProblemList.txt:797]: I've created https://bugs.openjdk.org/browse/JDK-8313404 to extract the fix for jdk_foreign from this PR. It's cleaner, also for backporting, although it incurs some process overhead.

**Comment by RealCLanger** [comment]
> Are the DatagramChannel tests passing on this platform? DatagramChannel supports multicasting.

We currently exclude these two tests on AIX in our private exclude list:
java/nio/channels/DatagramChannel/AfterDisconnect.java
java/nio/channels/DatagramChannel/SendReceiveMaxSize.java

We should probably add them here, too.

**Comment by AlanBateman** [comment]
> We should probably add them here, too.

java/nio/channels/DatagramChannel/SendReceiveMaxSize.java doesn't do multicasting but maybe it is excluded on that platform for some other reason? There are several DatagramChannel tests that do exercise multicast and I suspect they will fail on platforms that don't allow dual IPv4/IPv6 sockets to join IPv4 multicast groups.

**Comment by RealCLanger** [comment]
> > We should probably add them here, too.
> 
> java/nio/channels/DatagramChannel/SendReceiveMaxSize.java doesn't do multicasting but maybe it is excluded on that platform for some other reason? There are several DatagramChannel tests that do exercise multicast and I suspect they will fail on platforms that don't allow dual IPv4/IPv6 sockets to join IPv4 multicast groups.

I checked. SendReceiveMaxSize times out, probably for some other reason. So it should not be excluded here. But AfterDisconnect fails due to the IPv4/IPv6 issue, so this fits here.

**Comment by TOatGithub** [comment]
I can confirm this and will add DatagramChannel soon. SendReceiveMaxSize never returns from a native call to recvfrom...

**Comment by AlanBateman** [comment]
> I checked. SendReceiveMaxSize times out, probably for some other reason. So it should not be excluded here. But AfterDisconnect fails due to the IPv4/IPv6 issue, so this fits here.

That's fine. The tests that I expected to see listed are tests such as BasicMulticastTests, MulticastSendReceiveTests, AdaptorMulticasting and several others but maybe they are passing on this platform.

File: test/jdk/ProblemList.txt (hunk 1 above)
Comment (RealCLanger) [reply test/jdk/ProblemList.txt:799]: As discussed, please remove this part of the change.

**Review by RealCLanger — APPROVED** [review]
Looks good now.

**Comment by aixtools** [comment]
fyi: iirc, sendmaxsize stuff fails when it exceeds 4G in /tmp, asm by default, that is the size of /tmp on AIX systems configured by the ansible playbooks. There can be other issues - also resolved with the correct AIX settings (iirc, no settings for maxbuf, and max tcp and udp buffer sizes.

imho: correct to include the transfer size tests, but when they became 2x 2.1 G (which becomes 4.2G) - these tests failed until /tmp was increased to 5G.

Unlike many Linux systems "rootvg" on AIX is several interdependently sized filesystems.

**Comment by RealCLanger** [comment]
Yes, the tests that we didn't exclude here seem to work. There are a few network related tests still excluded on AIX in SAP's test system but these seem to have different symptoms. We're working on analyzing these and if necessary report the issues in JBS.

**Pull request closed without merging**